osv/models.py

```diff
@@ -332,18 +332,25 @@ def _tokenize(self, value):
     value_lower = value.lower()
     return re.split(r'\W+', value_lower) + [value_lower]
 
+  def _normalize(self, name):
+    """Normalize package name per https://peps.python.org/pep-0503/#normalized-names"""
+
+    return re.sub(r"[-_.]+", "-", name).lower()
+
   def _pre_put_hook(self):
     """Pre-put hook for populating search indices."""
     search_indices = set()
 
     search_indices.update(self._tokenize(self.id()))
 
-    for affected_package in self.affected_packages:
+    for pkg in self.affected_packages:
       # Set PURL if it wasn't provided.
-      if not affected_package.package.purl:
-        affected_package.package.purl = purl_helpers.package_to_purl(
+      if not pkg.package.purl:
```

Comment:
This should go after your normalization, as otherwise the PURL won't get the correct normalized value.

Reply:
Ah, good catch, thanks. Addressed in 47438b76ae7c3b24ee97d85478bc01325d888bb6